src/geom/SipApproximation.cc

```diff
@@ -33,53 +33,67 @@ namespace poly = lsst::geom::polynomials;
 
 namespace {
 
-std::pair<poly::PolynomialFunction2dYX, poly::PolynomialFunction2dYX> fitSipOneDirection(
+std::pair<poly::PolynomialFunction2dYX, poly::PolynomialFunction2dYX> fitPolynomial(
     int order,
-    lsst::geom::Box2D const & box,
     double svdThreshold,
     std::vector<lsst::geom::Point2D> const & input,
     std::vector<lsst::geom::Point2D> const & output
 ) {
-    // The scaled polynomial basis evaluates polynomials after mapping the
-    // input coordinates from the given box to [-1, 1]x[-1, 1] (for numerical
-    // stability).
-    auto basis = poly::ScaledPolynomialBasis2dYX(order, box);
+    auto basis = poly::PolynomialBasis2dYX(order);
     auto workspace = basis.makeWorkspace();
-    Eigen::MatrixXd matrix = Eigen::MatrixXd::Zero(input.size(), basis.size());
+    // Since we want to null the constant terms in the polynomial, we chop off
+    // the first column from the matrix we take the SVD of, and then set those
+    // coefficients to zero in the result.
+    Eigen::MatrixXd matrix = Eigen::MatrixXd::Zero(input.size(), basis.size() - 1);
     Eigen::VectorXd xRhs(input.size());
     Eigen::VectorXd yRhs(input.size());
+    Eigen::VectorXd tmp(basis.size());
     for (int i = 0; i < matrix.rows(); ++i) {
-        basis.fill(input[i], matrix.row(i), workspace);
-        auto rhs = output[i] - input[i];
+        tmp.setZero();
+        basis.fill(input[i], tmp, workspace);
+        matrix.row(i) = tmp.tail(basis.size() - 1);
+        auto rhs = output[i];
         xRhs[i] = rhs.getX();
         yRhs[i] = rhs.getY();
     }
-    // Since we're not trying to null the zeroth- and first-order terms, the
-    // solution is just linear least squares, and we can do that with SVD.
     auto decomp = matrix.jacobiSvd(Eigen::ComputeThinU | Eigen::ComputeThinV);
     if (svdThreshold >= 0) {
         decomp.setThreshold(svdThreshold);
     }
-    auto scaledX = makeFunction2d(basis, decomp.solve(xRhs));
-    auto scaledY = makeFunction2d(basis, decomp.solve(yRhs));
-    // On return, we simplify the polynomials by moving the remapping transform
-    // into the coefficients themselves.
-    return std::make_pair(simplified(scaledX), simplified(scaledY));
+    Eigen::VectorXd xCoeff = Eigen::VectorXd::Zero(basis.size());
+    if (Eigen::isnan(xCoeff.array()).any()) {
```

Comment:
Is this NaN check in the right place? You have just set `xCoeff` to zeros, so it should never be Nan. Was this meant to be after `decomp.solve`? Same comment for the `yCoeff` below.